test/test_meta.py

```diff
@@ -1,657 +1,107 @@
-    # preserve storage_offset, e.g.)
-    torch.Tensor.as_strided,
-    # This one segfaults when you call it
-    torch.Tensor.type,
-    # We don't clone autograd history, so this will generally not work
-    torch.autograd.grad,
-    torch.Tensor.backward,
-    torch.Tensor.__deepcopy__,
-    # Don't do factories
-    torch.ones,
-    torch.full,
-    torch.empty,
-    torch.randperm,
-    torch.logspace,
-    torch.zeros,
-    torch.arange,
-    torch.vander,
-    torch.as_tensor,
-    torch.tensor,
-    torch.randn_like,
-    torch.sparse_csr_tensor,
-    torch._sparse_coo_tensor_unsafe,
-    torch._sparse_csr_tensor_unsafe,
-    torch._validate_sparse_csr_tensor_args,
+dtype_abbrs = {
+    torch.bfloat16: 'bf16',
+    torch.float64: 'f64',
+    torch.float32: 'f32',
+    torch.float16: 'f16',
+    torch.complex32: 'c32',
+    torch.complex64: 'c64',
+    torch.complex128: 'c128',
+    torch.int8: 'i8',
+    torch.int16: 'i16',
+    torch.int32: 'i32',
+    torch.int64: 'i64',
+    torch.bool: 'b8',
+    torch.uint8: 'u8',
 }
 
-# This is a __torch_function__ mode that, when enabled, interposes every
-# Torch API call and runs the operator as normal, and then reruns it
-# with meta inputs, and then checks that everything about the output agrees.
-# Most of the logic deals with faithfully replicating the original tensor
-# as a meta tensor, which is nontrivial because there are a lot of subsystems
-# that may potentially be exercised.
-#
-# That being said, this class is a little overkill for what it is doing in
-# this test file (since I could have just inlined __torch_function__ on the
-# OpInfo call, and OpInfos generally have very regular inputs), but it will be
-# useful for more comprehensive testing e.g., as seen in
-# https://github.com/pytorch/pytorch/pull/75994
-class MetaCrossRefMode(torch.overrides.TorchFunctionMode):
-    test_case: TestCase
-    run_excludes_anyway: bool
-
-    def __init__(self, test_case, *, run_excludes_anyway):
-        self.test_case = test_case
-        self.run_excludes_anyway = run_excludes_anyway
+def safe_is_leaf(t):
+    try:
+        return t.is_leaf
+    except RuntimeError:
+        # inference mode can trigger this
+        return False
 
-    def __torch_function__(self, func, types, args=(), kwargs=None):
-        kwargs = kwargs or {}
 
-        hit = 0
-        miss = 0
+# This is a class for converting multiple tensors into meta tensors which
+# share the same view/storage structure.  The operation model is you allocate
+# one of these, and then call it repeatedly on all the tensors you want to
+# convert.  It's important to use the same object for tensors you want to
+# share storage because this is how we correlate shared storages to the same
+# meta storages; similarly, it's important NOT to use the same object for
+# unrelated groups of tensors because this class will remember all the
+# tensors/storages its seen and therefore leak memory.
+class MetaConverter:
```

Comment:
is this worth factoring out outside of test? didn't we have a `#TODO: aliasing` somewhere previously ?

Reply:
Yeah probably! Since it's top level now easy enough to move